ioSPI/micrographs.py

```diff
@@ -0,0 +1,78 @@
+"""Read and write micrographs."""
+
+import os
+
+import h5py
+import mrcfile
+import numpy as np
+
+
+def populate_hdf5_with_dict(h5file, path, dic):
+    """Recursively save dictionary contents to group.
+
+    Parameters
+    ----------
+    h5file : h5py.File
+        .hdf5 file to write to.
+    path : str
+        Relative path to save dictionary contents.
+    dic : dict
+        Dictionary containing data.
+    """
+    for k, v in dic.items():
+        if isinstance(v, (np.ndarray, np.int64, np.float64, int, float, str, bytes)):
+            h5file[path + k] = v
+        elif isinstance(v, type(None)):
+            h5file[path + k] = str("None")
+        elif isinstance(v, dict):
+            populate_hdf5_with_dict(h5file, path + k + "/", v)
+        else:
+            raise ValueError("Cannot save %s type" % type(v))
+
+
+def read_micrograph_from_mrc(mrc_file):
+    """Return micrograph from an input .mrc file.
+
+    Parameters
+    ----------
+    mrc_file : str
+        File name for .mrc file to turn into micrograph
+    """
+    with mrcfile.open(mrc_file, "r", permissive=True) as mrc:
+        micrograph = mrc.data
+    if len(micrograph.shape) == 2:
+        micrograph = micrograph[np.newaxis, ...]
+    return micrograph
+
+
+def write_data_dict_to_hdf5(data_dict, h5_file):
```

Comment:
This should respect the convention:
- first argument is **where** we will write, and it is called "path"
- second argument is **what** we will write: ok to call it data_dict.